### PR TITLE
delete macros

### PR DESCRIPTION
**Pull request by sandlerr**
#345 @zendesk/samson @zendesk/runway

### CODE REVIEW

File: app/views/macros/edit.html.erb [hunk 1]
```diff
@@ -5,5 +5,16 @@
 <h1>Edit macro</h1>
 
 <section>
+  <div class="row">
+    <div class="col-lg-2 button-label">Actions</div>
+    <div class="col-lg-6">
+      <%= link_to 'Execute', execute_project_macro_path(@project, @macro), method: :post, class: 'btn btn-primary', 'data-confirm' => "Are you sure you have no unsaved changes and would like to execute '#{@macro.name}'?" %>
```
Comment (grosser) [inline app/views/macros/edit.html.erb]: you can do `data: {confirm: "xxxxx"}`

File: app/views/macros/edit.html.erb [hunk 2]
```diff
@@ -5,5 +5,16 @@
...
+      <% if current_user.is_admin? %>
+        <%= link_to project_macro_path(@project, @macro), class: 'btn btn-danger', method: :delete, data: { confirm: "Are you sure?" } do %>
```
Comment (grosser) [inline app/views/macros/edit.html.erb]: `link_to [@project, @macro]` will build the same path

Comment (grosser) [inline app/views/macros/edit.html.erb]: maybe a `link_to_delete` helper would remove some duplication here like the classes/glyphicon etc

**Comment by grosser** [comment]
:+1:

**Comment by sandlerr** [comment]
@grosser refactored delete buttons & links

File: app/views/admin/environments/index.html.erb [hunk 3]
```diff
@@ -21,7 +21,7 @@
             <td>
               <%= link_to "Edit", edit_admin_environment_path(env) %>
               |
-              <%= link_to "Delete", admin_environment_path(env), method: :delete, data: { confirm: "Are you sure?" } %>
+              <%= link_to_delete(admin_environment_path(env)) %>
```
Comment (grosser) [inline app/views/admin/environments/index.html.erb]: `[:admin, env]` would also work :)

File: app/views/macros/edit.html.erb [hunk 4]
```diff
@@ -5,5 +5,14 @@
 <h1>Edit macro</h1>
 
 <section>
+  <div class="row">
+    <div class="col-lg-2 button-label">Actions</div>
+    <div class="col-lg-6">
+      <%= link_to 'Execute', execute_project_macro_path(@project, @macro), method: :post, class: 'btn btn-primary', data: { confirm: "Are you sure you have no unsaved changes and would like to execute '#{@macro.name}'?" } %>
+      <% if current_user.is_admin? %>
+        <%= link_to_delete_button([@project, @macro]) %>
```
Comment (grosser) [inline app/views/macros/edit.html.erb]: the authorization logic should ideally not be duplicated -> `current_user.can?(:destroy, @macro)` and use it in both places

File: app/views/admin/environments/index.html.erb (hunk 3 above)
Comment (sandlerr) [inline app/views/admin/environments/index.html.erb]: Couldn't find any docs so had to check the source, this magic comes from `url_for` which is also used in redirects. I'll go through and make these as consistent as I can.

Comment (sandlerr) [inline app/views/admin/environments/index.html.erb]: avoiding adding any `options` [here](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/routing/polymorphic_routes.rb#L99) so just updating links to `#index`

**Comment by grosser** [comment]
:+1:

**Comment by sandlerr** [comment]
cc @zendesk/samson now I've done it I'm not sure about it. This PR will make it more dry and concise

File: app/views/macros/edit.html.erb (hunk 4 above)
Comment (sandlerr) [inline app/views/macros/edit.html.erb]: we have `.can#{do_stuff}` all over the place in our models and view helpers. I'll fix that in another PR

File: app/helpers/deploys_helper.rb [hunk 5]
```diff
@@ -86,6 +86,10 @@ def stages_select_options
     end
   end
 
+  def stop_button(classes = 'btn btn-danger btn-xl')
```
Comment (grosser) [inline app/helpers/deploys_helper.rb]: I'd expect the first argument to be an options hash

**Comment by grosser** [comment]
I like the dry-ness, also less things to type :)

**Pull request merged**